proto/src/determined/api/v1/api.proto

```diff
@@ -509,6 +509,78 @@ service Determined {
       tags: "Internal"
     };
   }
+  // Get the absolute training length until the searcher needs the trial to
+  // report a validation.
+  rpc GetTrialSearcherTrainUntil(GetTrialSearcherTrainUntilRequest)
+      returns (GetTrialSearcherTrainUntilResponse) {
+    option (google.api.http) = {
+      get: "/api/v1/trials/{trial_id}/searcher/train_until"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+  // Reports to the searcher that the trial has completed the current
+  // requested amount of training with the given searcher validation
+  // metric.
+  rpc ReportTrialSearcherValidation(ReportTrialSearcherValidationRequest)
+      returns (ReportTrialSearcherValidationResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/trials/{trial_id}/searcher/report_validation"
```

Comment:
```suggestion
      post: "/api/v1/trials/{trial_id}/validation"
```

Reply:
ah, see, well this one is different, because we're going to have separate APIs to report a searcher validation and to report validation metrics just to be persisted.

Reply:
okay. I actually was lazy. This comment contains two things. One thing is the searcher portion of the path. The other thing is that `report_validation` isn't really restful pattern in API design unless we don't care about the restful design general practice.

Reply:
> The other thing is that report_validation isn't really restful pattern in API design unless we don't care about the restful design general practice.

What precisely do you mean by restful pattern? Whatever you mean, I just meant it to be a plain old RPC. I had CQRS in mind a bit, with RPCs that affect the state of the system and queries that retrieve it. I do like the REST pattern, i just dont think it fits everything perfectly (some how k8s has made it work though, I guess that's a good counterexample that it can be made to literally fit anything lol).

Reply:
talked offline. we will use the REST pattern.